src/DGtal/io/readers/ITKReader.ih

```diff
@@ -188,8 +188,17 @@ namespace DGtal {
       throw IOException();
     }
     
-    const TypeDGtalImage image( itk_image );
-
+    TypeDGtalImage image( itk_image );
+    if (shiftDomain)
+    {
+      auto c = itk_image->GetOrigin();
+      typename Domain::Point p;
+      for (unsigned int k = 0; k < dimension; k++)
+      {
+        p[k]=c[k];
```

Comment:
I would put a static_cast here, to be explicit about the change of type.
And the limitations that an origin of 0.5, would not change the domain.